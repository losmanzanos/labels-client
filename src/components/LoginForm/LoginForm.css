.LoginForm {
  width: 100%;
  max-width: 200px;
  margin: auto;
}
