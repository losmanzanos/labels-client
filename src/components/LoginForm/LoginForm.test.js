import renderer from "react-test-renderer";

//Smoke Test

// make React available
import React from "react";

// make the ReactDOM available, necessary for rendering the component
import ReactDOM from "react-dom";

// make the App component available
import LoginForm from "./LoginForm";

// this is the test case
it("renders without crashing", () => {
  // first create a DOM element to render the component into
  const div = document.createElement("div");

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<LoginForm />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

//Snapshot

it("renders the UI as expected", () => {
  const tree = renderer.create(<LoginForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
