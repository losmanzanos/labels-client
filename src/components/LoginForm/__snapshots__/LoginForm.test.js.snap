// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name"
  >
    <label
      htmlFor="LoginForm__user_name"
    >
      Username
    </label>
    <input
      className="Input "
      id="LoginForm__user_name"
      name="user_name"
      required={true}
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <input
      className="Input "
      id="LoginForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="Button "
    id="login"
    type="submit"
  >
    Login
  </button>
</form>
`;
